src/participant/prompter.ts

```diff
@@ -699,6 +715,102 @@ export class PrompterParticipant {
             }
         }
     }
+
+    private async _handleSuggestByRules(
+        request: vscode.ChatRequest,
+        context: vscode.ChatContext,
+        stream: vscode.ChatResponseStream,
+        token: vscode.CancellationToken
+    ) {
+        this._sendCommandStartMessage(
+            stream,
+            'suggest a new prompt using rules suggested by OpenAI'
+        );
+
+        const prompt = await this._getPrompt(stream);
+        if (prompt) {
+            stream.markdown('\n\n');
+            const suggestion = await suggestImprovement(prompt);
+            // Render the results
+            stream.markdown('** Suggestion Results**:\n\n');
+            stream.markdown(this.handleSuggestImprovement(suggestion));
+        }
+        return { metadata: { command: 'suggest-by-rules' } };
+    }
+
+    private handleSuggestImprovement(response: JSONSchemaObject): string {
+        let return_message = '';
+        if (response.error) {
+            return_message += 'Error: ';
+            return_message += response.error as string;
+            return_message += '\n\n';
+        } else if (response.suggestion) {
+            return_message += ' **Suggestion:** \n\n';
+            return_message += response.suggestion as string;
+            return_message += '\n\n';
+        } else {
+            return_message += 'Response from API was poorly formatted.';
+            return_message += '\n\n';
+        }
+
+        return return_message;
+    }
+
+    _sendCommandStartMessage(
```

Comment:
Great addition! I will integrate that into the other models as well once the PR is merged.